File: java/RecursiveAlgorithms/src/com/sethholloway/Fibonacci.java
package com.sethholloway;

public class Fibonacci {
	private long sequence = 0;
	
	public Fibonacci() {
	}
	
	public Fibonacci(int n) {
		sequence = fib(n);
	}
	
	public long getSequence() {
		return sequence;
	}
	
	public static long fib(int n) {
		if (n <= 2) {
			return 1;
		} else {
			return fib(n-1) + fib(n-2);
		}
	}

}

File: go/fibonacci/fibonacci.go
package main

import (
    "fmt"
    "math/big"
)

func intfibonacci() func() int64 {
    first, second := int64(0), int64(1)
    return func() int64 {
        first, second = second, first + second
        return first
    }
}

// fibonacci generator
func fibonacci() func() *big.Int {
    first, second := big.NewInt(0), big.NewInt(1)
    return func() *big.Int {
        first, second = second, first.Add(first, second)
        return first
    }
}

func main() {
    f := fibonacci()
    for i := 0; i < 100; i++ {
        fmt.Printf("fib(%v) = %v\n", i, f())
    }
}
